src/function/mod.rs

```diff
@@ -34,12 +34,16 @@ impl<TS: TypeSystem> Function<TS> {
         }
 
         for expr in self.expressions.iter().take(self.expressions.len() - 1) {
-            if let Err(FreightError::Return { target }) = engine.evaluate(expr, args, captured) {
-                if target == self.return_target {
-                    return Ok(std::mem::take(&mut engine.return_value));
-                } else {
-                    return Err(FreightError::Return { target });
+            match engine.evaluate(expr, args, captured) {
+                Err(FreightError::Return { target }) => {
+                    if target == self.return_target {
+                        return Ok(std::mem::take(&mut engine.return_value));
+                    } else {
+                        return Err(FreightError::Return { target });
+                    }
                 }
+                Err(e) => return Err(e),
```

Comment:
This is a major change to how functions work, making them automatically return the first thing that evaluates to an error - I think this is something we should discuss this in a design meeting, because it introduces hidden control flow that's not necessarily obvious.

Reply:
what if just `IncorrectArgumentCount` gets returned?

Reply:
That's even more inconsistent then

Reply:
not really, especially because apparently that already happens if the function isn't a fixed arg count

Reply:
my original solution to this was a function passed from the dependent project (fender) I didn't go with it cause that felt more inconsistent than this because I couldn't figure out where other errors were coming from

Reply:
> not really, especially because apparently that already happens if the function isn't a fixed arg count

Then we should fix that, I think

Reply:
> my original solution to this was a function passed from the dependent project (fender) I didn't go with it cause that felt more inconsistent than this because I couldn't figure out where other errors were coming from

I don't understand what you're describing

Reply:
fender handles an error being raised instead of freight just silently doing nothing